cartridges/app_storefront_pdp_extend/cartridge/templates/default/product/components/addToCartProduct.isml

```diff
@@ -0,0 +1,30 @@
+
+<isscript>
+
+</isscript>
+
+
+
+
+<div class="row cart-and-ipay">
+
+    <div id="addToCartButton" class="col-sm-12 ${!ProductMgr.getProduct(product.id).availabilityModel.inStock ? "d-none" : ""} ">
+        <input type="hidden" class="add-to-cart-url" value="${pdict.addToCartUrl}">
+
+        <button class="add-to-cart btn btn-primary" data-toggle="modal" data-target="#chooseBonusProductModal"
+            data-pid="${product.id}" ${!product.readyToOrder || !product.available ? "disabled" : ""}>
+            <i class="fa fa-shopping-bag"></i>
+            ${Resource.msg('button.addtocart', 'common', null)}
+        </button>
+
+    </div>
+
+    <h1></h1>
+
+    <div class="${ProductMgr.getProduct(product.id).availabilityModel.inStock ? "d-none" : ""}">
```

Comment:
Availability should be from product model. Do not call the API from the template.